utils/sites.js

```diff
@@ -207,12 +207,12 @@ const updateParticipantData = async (req, res, authObj) => {
     if(req.body.data.length > 100) return res.status(400).json(getResponseJSON('Bad request. Data contains more than acceptable limit of 100 records.', 400));
 
     const dataArray = req.body.data;
-    const primaryIdentifiers = ['token', 'pin', 'Connect_ID', 'state.uid'];
 
     let responseArray = [];
     let error = false;
 
-    for(let dataObj of dataArray) {
+    // Iterate dataArray, Fetching each participant record from firestore by token.
+    participantLoop: for(let dataObj of dataArray) {
```

Comment:
does this just label the loop?

Reply:
Yes, this creates a jump target from the inner loop in cancer occurrence validation

```javascript
if (dataObj[fieldMapping.cancerOccurrence]) {
                        const existingOccurrences = docData[fieldMapping.cancerOccurrence] || [];
                        const requiredOccurrenceRules = Object.keys(rules).filter(key => key.startsWith(fieldMapping.cancerOccurrence.toString()) && rules[key].required);
                        const occurrenceResult = handleCancerOccurrences(flatDataObj, dataObj, existingOccurrences, requiredOccurrenceRules);
                        if (occurrenceResult.error) {
                            error = true;
                            responseArray.push({'Invalid Request': {'Token': participantToken, 'Errors': occurrenceResult.message}});
                            continue participantLoop; // <-- Jump target here
                        }
                    }
```

Without this, I believe the error could be returned from cancer occurrence validation and this loop would continue, potentially allowing the request to validate even though it shouldn't. With the jump target, we return the error for this participant's request and skip to the next iteration of participant loop, outside of the current loop.